File: implementations/kotlin-springboot/harness/test/testdata/repository-naming/good/domain/AccountRepository.kt
package com.example.accountservice.account.domain

interface AccountRepository {
    fun findAccounts(query: AccountFindQuery): Pair<List<Account>, Long>

    fun saveAccount(account: Account)

    fun deleteAccount(accountId: String)

    // A domain query method that doesn't start with find/save/delete/count, like
    // hasTransactionWithReference, must not trip the blocklist (verifies false-positive avoidance).
    fun hasTransactionWithReference(referenceId: String): Boolean
}

File: implementations/nestjs/harness/tests/fixtures/repository-naming/good/src/account/domain/account-repository.ts
import { Account } from './account'

export abstract class AccountRepository {
  abstract findAccounts(query: {
    readonly take: number
    readonly page: number
    readonly accountId?: string
    readonly ownerId?: string
  }): Promise<{ accounts: Account[]; count: number }>

  abstract saveAccount(account: Account): Promise<void>

  abstract deleteAccount(accountId: string): Promise<void>

  abstract hasTransactionWithReference(referenceId: string): Promise<boolean>
}
